apps/web/src/components/commands/tokens/Tokens.tsx

```diff
@@ -0,0 +1,73 @@
+import { DataPill } from "../../DataPill";
+import { useQuery } from "react-query";
+import { PortalsResponse, Token } from "./types";
+import { findToken } from "./helpers";
+import Image from "next/image";
+
+export type TokenWidgetProps = {
+  property: "price";
+  chainName: string;
+  token: string;
+  platforms: "native" | "basic";
+};
+
+export const TokenWidget = ({ property, ...props }: TokenWidgetProps) => {
+  switch (property) {
+    case "price":
+      return <TokenPriceWidget {...props} />;
+    default:
+      return null;
+  }
+};
+
+const TokenPriceWidget = ({
+  chainName,
+  token,
+  platforms,
+}: Pick<TokenWidgetProps, "chainName" | "token" | "platforms">) => {
+  const query = useQuery(
+    ["TOKEN", "PRICE", chainName, token, platforms],
+    async () => {
+      if (token.toLowerCase().trim() === "eth") {
+        platforms = "native";
```

Comment:
Can reassigning this prop cause any issues?

Reply:
I thought about it. Only problem might be if we switch back. There is no else.